core/src/main/java/brooklyn/location/access/PortForwardManagerImpl.java

```diff
@@ -78,7 +78,9 @@
     private static final Logger log = LoggerFactory.getLogger(PortForwardManagerImpl.class);
     
     protected final Map<String,PortMapping> mappings = new LinkedHashMap<String,PortMapping>();
-    
+
+    private Map<AssociationListener, Predicate<? super AssociationMetadata>> associationListeners = new ConcurrentHashMap<AssociationListener, Predicate<? super AssociationMetadata>>();
```

Comment:
Declare the field `final`. It makes reasoning about the code simpler for folk reading or modifying it later.